### DISCUSSION TITLE
Stanford Digital Repository report on child sexual abuse material (CSAM) on Mastodon

### DISCUSSION THREAD

**Discussion by Yekrats** [mastodon/mastodon#26148] [General]
Standord Cyber Policy Center released a report about Mastodon today, saying that it is rife with child exploitation, in relation to images and hashtags that some instances allow. https://cyber.fsi.stanford.edu/io/news/addressing-child-exploitation-federated-social-media As a result, the Washington Post has already released a negative story about Mastodon and the Fediverse today.

Their report is worth checking out, as they come up with some ideas of how to integrate tools like Microsoft's PhotoDNA and/or other tools when a user uploads a photo.

I know that most instances have some sort of federated blocking system. The instances that I use have implemented the [Tier 0 blocklist](https://writer.oliphant.social/oliphant/the-blocklist-algorithm) which walls off most of the bad servers where one could find that crap. I think most of it is addressed pretty quickly, and walled off. We can't be responsible for bad actors using good software. However, we can give our good actors the tools to stop and delete bad content, and make it easier to relegate the garden inside the wall.

What can be done about this? 

- Can an admin set up automatic blocks or flags for certain hashtags?
- Can Mastodon integrate some automatic checking of images, or tools that admins can use to take care of this issue?

**Comment by renchap** [comment]
This is a complex issue and is currently being discussed a lot of how to implement it, while keeping things decentralized and privacy-centric.

I recently wrote [an article about how I envision it](https://renchap.com/blog/post/evolving_mastodon_trust_and_safety/), but nothing has been decided yet.

**Comment by i5heu** [comment]
Are there any news about this topic?

A integration of https://www.microsoft.com/en-us/photodna would be sufficient for at least the short time and would provide a better security for the community and admins.
